src/uct/ib/base/ib_iface.c

```diff
@@ -385,25 +403,27 @@ void uct_ib_iface_address_pack(uct_ib_iface_t *iface, uct_ib_address_t *ib_addr)
 {
     uct_ib_address_pack_params_t params;
 
-    params.flags     = uct_ib_iface_address_pack_flags(iface);
-    params.gid       = &iface->gid_info.gid;
-    params.lid       = uct_ib_iface_port_attr(iface)->lid;
-    params.roce_info = &iface->gid_info.roce_info;
+    params.flags      = uct_ib_iface_address_pack_flags(iface);
+    params.gid        = &iface->gid_info.gid;
+    params.lid        = uct_ib_iface_port_attr(iface)->lid;
+    params.roce_info  = &iface->gid_info.roce_info;
     /* to suppress gcc 4.3.4 warning */
-    params.path_mtu  = UCT_IB_ADDRESS_INVALID_PATH_MTU;
-    params.gid_index = UCT_IB_ADDRESS_INVALID_GID_INDEX;
+    params.path_mtu   = UCT_IB_ADDRESS_INVALID_PATH_MTU;
+    params.gid_index  = UCT_IB_ADDRESS_INVALID_GID_INDEX;
+    params.pkey_value = iface->pkey_value;
     uct_ib_address_pack(&params, ib_addr);
 }
 
 void uct_ib_address_unpack(const uct_ib_address_t *ib_addr, uint16_t *lid,
                            union ibv_gid *gid, uint8_t *gid_index,
-                           enum ibv_mtu *path_mtu)
+                           enum ibv_mtu *path_mtu, uint16_t *pkey_value)
 {
     const void *ptr = ib_addr + 1;
 
-    *lid       = 0;
-    *gid_index = UCT_IB_ADDRESS_INVALID_GID_INDEX;
-    *path_mtu  = UCT_IB_ADDRESS_INVALID_PATH_MTU;
+    *lid        = 0;
+    *gid_index  = UCT_IB_ADDRESS_INVALID_GID_INDEX;
+    *path_mtu   = UCT_IB_ADDRESS_INVALID_PATH_MTU;
+    *pkey_value = UCT_IB_ADDRESS_PKEY_VALUE_DEFAULT;
```

Comment:
UCT_IB_ADDRESS_PKEY_VALUE_DEFAULT -> UCT_IB_ADDRESS_DEFAULT_PKEY_VALUE
also maybe rename pkey_value->pkey in all places (pkey would mean the value and "pkey_index" would be the index when applicable)